### PR TITLE
temporal-cli: combine temporal-cli and tctl

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/temporal-cli/default.nix [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
+      hash = "sha256-yQnFw3uYGKrTevGFVZNgkWwKCCWiGy0qwJJOmnMpTJQ=";
+    };
+
+    vendorHash = "sha256-ld59ADHnlgsCA2mzVhdq6Vb2aa9rApvFxs3NpHiCKxo=";
+
+    nativeBuildInputs = [ installShellFiles ];
+
+    excludedPackages = [ "./cmd/docgen" ];
+
+    ldflags = [ "-s" "-w" ];
```
Comment (madjam002) [inline pkgs/applications/networking/cluster/temporal-cli/default.nix:21]: Also needs `"-X=github.com/temporalio/cli/headers.Version=${version}"`
for certain commands to function e.g `temporal workflow describe`

File: pkgs/applications/networking/cluster/temporal-cli/default.nix [hunk 2]
```diff
@@ -1,35 +1,83 @@
...
+
+    postInstall = ''
+      installShellCompletion --cmd tctl \
+        --bash <($out/bin/tctl completion bash) \
+        --zsh <($out/bin/tctl completion zsh)
+    '';
   };
+in
+symlinkJoin {
+  name = "temporal-cli-${tctl.version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/temporal-cli/default.nix:69]:
```suggestion
  pname = "temporal-cli";
  inherit (tctl) version;
  name = "${pname}-${version}";
```

Comment (aaronjheng) [reply pkgs/applications/networking/cluster/temporal-cli/default.nix:69]: Isn't `name` enough for `symlinkJoin`?

Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/temporal-cli/default.nix:69]: Yes but if we are treating temporal-cli like a package then it is expected that it has pname+version.

Comment (aaronjheng) [reply pkgs/applications/networking/cluster/temporal-cli/default.nix]: Done.

**Pull request merged**